src/services/findAllReferences.ts

```diff
@@ -840,6 +840,14 @@ namespace ts.FindAllReferences {
                     // import("foo") with no qualifier will reference the `export =` of the module, which may be referenced anyway.
                     return nodeEntry(reference.literal);
                 }
+                else if (reference.kind === "implicit") {
+                    // Return either: The first JSX node in the file, the first statement of the file, or the whole file if neither of those exist
+                    const range = forEachChildRecursively(
+                        reference.referencingFile,
+                        n => isJsxElement(n) || isJsxSelfClosingElement(n) || isJsxFragment(n) ? n : undefined

```

Comment:
Hm - in very large files, this could be a full tree-walk. I know it's a *transform* flag, but can you use `TransformFlags.ContainsJsx` to make this a little more clever?

Reply:
Done